CRM/Core/BAO/Email.php

```diff
@@ -324,10 +325,10 @@ public static function getFromEmail() {
         if (!empty($emailVal['is_primary'])) {
           $fromEmailHtml .= ' ' . ts('(preferred)');
         }
-        $fromEmailValues[$emailId] = $fromEmailHtml;
+        $contactFromEmails[$fromEmail] = $fromEmailHtml;
```

Comment:
The changed keying here is at least then consistent with https://github.com/civicrm/civicrm-core/pull/11905/files#diff-dd5148d8b660fec931daeee2d6c2d201R290 which @Stoob reports on the other PR that it works